common/OpTestHMC.py

```diff
@@ -417,12 +417,23 @@ def change_proc_mode(self, proc_mode, sharing_mode, min_proc_units, desired_proc
         :param overcommit_ratio: overcommit ratio can be 1 to 5 for ideal cases
         '''
         if proc_mode == 'shared':
+            '''
+            Get the maximum configured virtual procs
+            '''
+            v_max_proc = 0
+            max_virtual_proc = self.run_command("lshwres -m %s -r proc --level sys -F curr_sys_virtual_procs" % (self.mg_system))
+            max_virtual_proc = int(max_virtual_proc[0])
+            if 2*int(max_proc_units) > max_virtual_proc:
+                v_max_proc = max_virtual_proc
+            else:
+                v_max_proc = 2*int(max_proc_units)
+
             self.set_lpar_cfg("proc_mode=shared,sharing_mode=%s,min_proc_units=%s,max_proc_units=%s,"
                               "desired_proc_units=%s,min_procs=%s,desired_procs=%s,max_procs=%s,"
                               "min_mem=%s,desired_mem=%s,max_mem=%s" %
-                              (sharing_mode, overcommit_ratio*int(min_proc_units), max_proc_units, overcommit_ratio*int(desired_proc_units),
-                               int(min_proc_units), 2*int(desired_proc_units),
-                               2*int(max_proc_units), min_memory, desired_memory, max_memory))
+                              (sharing_mode, min_proc_units, max_proc_units, desired_proc_units,
+                                  int(min_proc_units),  overcommit_ratio*int(desired_proc_units), v_max_proc,
```

Comment:
@SamirMulani we do not need to have `overcommit_ratio` as a multiplier for min and max (instead of 2 ) as well

Reply:
@PraveenPenguin The `overcommit_ratio` should not be used as a multiplier for the min and max values. For the min, it should be equal to the user input, while for the max, it should either follow resource count or mac vproc value.